fmt/src/selection.rs

```diff
@@ -182,15 +180,45 @@ fn select_files_with_git(
         builder.build().context(SelectWithIgnoreSnafu)?
     };
 
-    let mut it = WalkDir::new(basedir).into_iter();
+    let basedir = basedir
+        .canonicalize()
+        .with_context(|_| ResolveAbsolutePathSnafu {
+            path: basedir.display().to_string(),
+        })?;
+    let mut it = WalkDir::new(basedir).follow_links(false).into_iter();
+
+    let workdir = repo
+        .work_dir()
+        .context(GixPathNotFountSnafu { path: "workdir" })?;
+    let workdir = workdir
+        .canonicalize()
+        .with_context(|_| ResolveAbsolutePathSnafu {
+            path: workdir.display().to_string(),
+        })?;
+    let worktree = repo
+        .worktree()
+        .context(GixPathNotFountSnafu { path: "worktree" })?;
+    let mut excludes = worktree
+        .excludes(None)
+        .map_err(Box::new)
+        .context(GixExcludeOpSnafu)?;
+
     while let Some(entry) = it.next() {
         let entry = entry.context(TraverseDirSnafu)?;
         let path = entry.path();
         let file_type = entry.file_type();
+
+        let rela_path = path
+            .strip_prefix(&workdir)
+            .expect("git repository encloses iteration");
+        let platform = excludes
+            .at_path(rela_path, Some(file_type.is_dir()))
+            .context(GixCheckExcludeOpSnafu)?;
+
         if file_type.is_symlink() {
             debug!("skip symlink: {:?}", path);
         } else if file_type.is_dir() {
-            if git_helper.ignored(path, true)? {
+            if platform.is_excluded() {
```

Comment:
This call also performs work, as it now has to search through the initialized stack of `.gitignore` patterns to find a match.

Reply:
I don't get it clear. Isn't this change part of the cascading of 

```rust
let platform = excludes.at_path(rela_path, Some(file_type.is_dir()))
```

above? In the previous code, it wrote:

```rust
        let platform = attrs
            .at_path(at_path, Some(is_dir))
            .context(GixCheckExcludeOpSnafu)?;
        Ok(platform.is_excluded())
```

The construction of `attrs` is different, but the rest is the same.

Reply:
Yes, what I tried to say is that the `is_excluded()` information isn't cached, but it is computed during the call. It was probably unnecessary to point that out (but I found it interesting enough).

Reply:
Good to know :D